docs/use_cases/knowledge_graph_embedding.md

```diff
@@ -0,0 +1,228 @@
+
+You can find all the code for this demonstration in [this](https://drive.google.com/file/d/1G3tJ6Nn_6hKZ8HZGpx8OHpWwGqp_sQtF/view?usp=sharing) notebook.
+
+## Comparing KGE with LLM performance on a large Knowledge Graph
+
+In this section, we're comparing the performance of Knowledge Graph Embeddings (KGE) and Language Models (LLMs) on a dataset called ogbl-wikikg2. This dataset is drawn from Wikidata and includes 2.5 million unique entities, 535 types of relations, and 17.1 million fact triplets. We'll evaluate their performance using hit rates (the ratio of correct answers), following the guidelines provided [here](https://ogb.stanford.edu/docs/linkprop/#ogbl-wikikg2).
+
+Our approach involved creating textual representations for each node within the graph. We did this by crafting sentences that describe their connections, like this: "[node] [relation1] [neighbor1], [neighbor2]. [node] [relation2] [neighbor3], [neighbor4]. ..." These textual representations were then fed into a LLM – specifically, the `BAAI/bge-base-en-v1.5` model available on [HuggingFace](https://huggingface.co/BAAI/bge-base-en-v1.5). The resulting embeddings from this process served as our node embeddings.
+
+For queries, we took a similar textual representation approach, creating descriptions of the query but omitting the specific entity in question. With these representations in hand, we utilized a dot product similarity to find and rank relevant answers.
+
+For the KGE algorithm, we employed DistMult with a 250-dimensional embedding space.
+
+
+### Results
+You can see the results on the Open Graph Benchmark query set in the table below:
+
+| metric/model  | Random | LLM | DistMult|
+| --- | --- | --- | --- |
+| HitRate@1 |  0.001 | 0.0055 | **0.065** |
+| HitRate@3 |  0.003 | 0.0154 | **0.150** |
+| HitRate@10 |  0.010 | 0.0436 | **0.307** |
+
+The table above clearly shows that LLM performs three times better than the method of randomly ordering nodes. It is also clear that KGE significantly outperforms LLM, with hit rates almost ten times higher. Notably, DistMult can find the correct answer on its first attempt more frequently than LLM can in ten attempts. It's important to point out that we used only 250-dimensional embeddings with DistMult, while LLM outputs 768-dimensional vectors, which makes the performance difference even more noticeable.
+
+These results strongly support that KGE is more suitable than LLMs for tasks where relational information is super important.
+
+## Limitations
+
+One reason why the LLM approach might struggle with performance is due to the formulation used, where each node is mapped to a sequence of sentences describing its connections. This method tends to overload the input text with an extensive amount of information for a single node. LLMs are typically not trained to handle such broad and diverse information within a single context; their strength lies in processing more focused and specific textual information.
```

Comment:
I think you should rephrase this. It's not that we cheated, it's that relational data is hard to be described to and grasped by an LLM. That is an inherent weakness of the approach, hence we should use KGEs.